### PR TITLE
Add: Policy to configure external watchdog for cf-execd

### PR DESCRIPTION
**Pull request by nickanderson**
Ref: https://dev.cfengine.com/issues/7324

### CODE REVIEW

**Comment by nickanderson** [comment]
@estenberg @tzz @kacfengine thoughts?

**Comment by kacfengine** [comment]
A couple of things:

* I think you should make this dependent on the existence of pgrep as well, since Linux is the only one that has it.
* This is new functionality, I think this should go into 3.8 only.

File: cfe_internal/core/watchdog/watchdog.cf [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+      "$(state)"
+        expression => "any";
+
+  vars:
+    any::
+
+      "cf_execd"
+        string => ifelse("windows", translatepath('"$(sys.bindir)/cf-execd.exe)"'),
+                         "$(sys.bindir)/cf-execd");
```
Comment (kacfengine) [inline cfe_internal/core/watchdog/watchdog.cf]: There is the ``$(sys.cf_execd)`` variable, which saves you from doing this.

File: cfe_internal/core/watchdog/watchdog.cf [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  files:
+
+    enabled.have_cron_d::
+
+      "$(cron_d_watchdog)"
+        create => "true";
+
+      "$(cron_d_watchdog)"
+        edit_defaults => empty,
```
Comment (kacfengine) [inline cfe_internal/core/watchdog/watchdog.cf]: Not needed for ``edit_template``, all templates start with an empty file. Only ``edit_line`` uses ``edit_defaults``.

**Comment by kacfengine** [comment]
Looks useful though!

**Comment by tzz** [comment]
Looks reasonable.  It should complain loudly if `/etc/cron.d` doesn't exist.

File: cfe_internal/core/watchdog/watchdog.cf (hunk 1 above)
Comment (nickanderson) [inline cfe_internal/core/watchdog/watchdog.cf]: thanks

File: cfe_internal/core/watchdog/watchdog.cf (hunk 2 above)
Comment (nickanderson) [inline cfe_internal/core/watchdog/watchdog.cf]: ack

**Comment by kacfengine** [comment]
New one with slightly tweaked Changelog in #516.

**Comment by kacfengine** [comment]
Merged #516.

**Pull request closed without merging**